go_util/util_test.go

```diff
@@ -85,3 +85,101 @@ links:
+  - product: firefox
+    url: https://bug.com/item
+    results:
+    - test: b.html
+      subtest: Something should happen
+      status: FAIL
+    - test: c.html
+`)
+	var metadata shared.Metadata
+	yaml.Unmarshal(fileInBytes, &metadata)
+
+	_, resultLinks := deleteMetadata("abc.html", metadata)
+
+	assert.Equal(t, 0, len(resultLinks))
+}
+
+func TestDeleteMetadata_DeletedOneLink(t *testing.T) {
+	fileInBytes := []byte(`
+links:
+  - product: chrome
+    url: https://external.com/item
+    results:
+    - test: a.html
+  - product: firefox
+    url: https://bug.com/item
+    results:
+    - test: b.html
+      subtest: Something should happen
+      status: FAIL
+    - test: c.html
+`)
+	var metadata shared.Metadata
+	yaml.Unmarshal(fileInBytes, &metadata)
+
+	resultMetadata, resultLinks := deleteMetadata("a.html", metadata)
+
+	assert.Equal(t, 1, len(resultLinks))
+	assert.Equal(t, "chrome", resultLinks[0].Product.BrowserName)
+	assert.Equal(t, "https://external.com/item", resultLinks[0].URL)
+	assert.Equal(t, 1, len(resultLinks[0].Results))
+	assert.Equal(t, "a.html", resultLinks[0].Results[0].TestPath)
+
+	assert.Equal(t, 1, len(resultMetadata.Links))
+	assert.Equal(t, "firefox", resultMetadata.Links[0].Product.BrowserName)
+	assert.Equal(t, "https://bug.com/item", resultMetadata.Links[0].URL)
+	assert.Equal(t, 2, len(resultMetadata.Links[0].Results))
+	assert.Equal(t, "b.html", resultMetadata.Links[0].Results[0].TestPath)
+	assert.Equal(t, "c.html", resultMetadata.Links[0].Results[1].TestPath)
+}
+
+func TestDeleteMetadata_DeletedMultipleLinks(t *testing.T) {
+	fileInBytes := []byte(`
+links:
+  - product: chrome
+    url: https://external.com/item
+    results:
+    - test: a.html
+  - product: firefox
+    url: https://bug.com/item
+    results:
+    - test: a.html
+      subtest: Something should happen
+      status: FAIL
+    - test: c.html
+`)
+	var metadata shared.Metadata
+	yaml.Unmarshal(fileInBytes, &metadata)
+
+	resultMetadata, resultLinks := deleteMetadata("a.html", metadata)
+
+	assert.Equal(t, 2, len(resultLinks))
+	assert.Equal(t, "chrome", resultLinks[0].Product.BrowserName)
+	assert.Equal(t, "https://external.com/item", resultLinks[0].URL)
+	assert.Equal(t, 1, len(resultLinks[0].Results))
+	assert.Equal(t, "a.html", resultLinks[0].Results[0].TestPath)
+	assert.Equal(t, "firefox", resultLinks[1].Product.BrowserName)
+	assert.Equal(t, "https://bug.com/item", resultLinks[1].URL)
+	assert.Equal(t, 1, len(resultLinks[0].Results))
+	assert.Equal(t, "a.html", resultLinks[0].Results[0].TestPath)
+
+	assert.Equal(t, 1, len(resultMetadata.Links))
+	assert.Equal(t, "firefox", resultMetadata.Links[0].Product.BrowserName)
+	assert.Equal(t, "https://bug.com/item", resultMetadata.Links[0].URL)
+	assert.Equal(t, 1, len(resultMetadata.Links[0].Results))
+	assert.Equal(t, "c.html", resultMetadata.Links[0].Results[0].TestPath)
+}
+
+func TestGetYMLFilePath(t *testing.T) {
+	assert.Equal(t, "abc/META.yml", getYMLFilePath("abc/abc"))
```

Comment:
```suggestion
	assert.Equal(t, "abc/META.yml", getYMLFilePath("abc/test.html"))
```

I assume this will also pass, but it makes it a bit more realistic.